### PR TITLE
included tumour dependencies for tumour-related specimen fields

### PR DESCRIPTION
**Pull request by hknahal**
This is related to https://github.com/icgc-argo/argo-dictionary/issues/62

### CODE REVIEW

File: schemas/specimen.json [hunk 1]
```diff
@@ -236,39 +243,52 @@
...
       "description": "Indicate a value, in decimals, that represents local response to cellular injury, marked by capillary dilatation, edema and leukocyte infiltration; clinically, inflammation is manifest by redness, heat, pain, swelling and loss of function, with the need to heal damaged tissue.",
       "valueType": "number",
       "meta": {
-        "core": true
+        "core": true,
+        "dependsOn": "sample_registration.tumour_normal_designation",
+        "note": "This field is only required if the specimen is tumour."
+      },
+      "restrictions": {
+        "required": true
```
Comment (rosibaj) [inline schemas/specimen.json]: If this is only required when tumor, then required :true should be removed here,, like the other fields.

File: schemas/specimen.json [hunk 2]
```diff
@@ -51,7 +51,9 @@
       "description": "Specify the tumour staging system used to assess the cancer at the time the tumour specimen was resected. Pathological classification is based on the clinical stage information (acquired before treatment) and supplemented/modified by operative findings and pathological evaluation of the resected specimen.",
       "valueType": "string",
       "meta": {
-        "core": true
+        "core": true,
+        "dependsOn": "sample_registration.tumour_normal_designation",
+        "note": "This field is only required if the specimen is tumour."
```
Comment (kcullion) [inline schemas/specimen.json]: @hknahal this looks good...maybe could say
This field is only required if the specimen is a tumour
or
This field is only required for tumour specimen

Comment (hknahal) [reply schemas/specimen.json]: @kcullion I included it in the "note" field (https://github.com/icgc-argo/argo-dictionary/blob/hn_56/schemas/specimen.json#L56). Are you suggesting it should be in the "description" instead?

Comment (kcullion) [reply schemas/specimen.json]: no sorry was just suggesting to add "a" (so a wording change)

Comment (kcullion) [reply schemas/specimen.json]: or does it make sense to say when specimen is tumour?

File: schemas/specimen.json (hunk 1 above)
Comment (hknahal) [reply schemas/specimen.json]: I removed the "required:true" where there is a dependency.

**Pull request merged**